### PR TITLE
Adds specialized formatter to histogram component.

### PR DESCRIPTION
**Pull request by bileschi**
* Motivation for features / changes
Using SI notation for histogram bins smaller than 1 causes confusion.  '500m' can mean many things, not just '0.5'.  Googlers see b/238492397

* Technical description of changes
Replaces tick formatter with a 3 way switch.  SI notation for small values, Exponential notation for large values, and natural notation for middle values.

* Screenshots of UI changes

before:  
----

image
image

after:
----

image
image

* Detailed steps to verify changes work correctly (as executed by you)
Ran TensorBoard standalone locally and compared across a very wide range of histogram bin bucket sizes.

* Alternate designs / implementations considered
It may be possible to update tensorboard/webapp/widgets/histogram/histogram_component.ts and re-use formatters from there.   Did not pursue as it would require extra customization there, and this was deemed sufficiently clear.

### CODE REVIEW

**Review by rileyajones — APPROVED** [review]
LGTM

File: tensorboard/webapp/widgets/histogram/histogram_component.ts [hunk 1]
```diff
@@ -76,6 +76,34 @@ export interface TooltipData {
...
+        return '0';
+    }
+    const absNum = Math.abs(x as number);
+    if (absNum >= LARGE_NUMBER){
+        return d3LargeFormatter(x);
+    }
+    if (absNum < SMALL_NUMBER){
+        return d3SmallFormatter(x);
+    }
+    return d3MiddleFormatter(x);
```
Comment (rileyajones) [inline tensorboard/webapp/widgets/histogram/histogram_component.ts:102]: This is very nitpicky but it's a little strange that this is ordered `LARGE_NUMBER`, `SMALL_NUMBER`, `MEDIUM_NUMBER`.

Perhaps consider gating around a series minimums.
```suggestion
    if (absNum >= LARGE_NUMBER){
        return d3LargeFormatter(x);
    }
    if (absNum >= SMALL_NUMBER){
        return d3MiddleFormatter(x);
    }
    return d3SmallFormatter(x);
```

If taking this suggestion you might also consider renaming `SMALL_NUMBER` to `MEDIUM_NUMBER`.

**Review by bmd3k — COMMENTED** [review]
Please add some tests if it is reasonable to do.

**Comment by bileschi** [comment]
Factored out the formatter and added tests.

**Review by bmd3k — COMMENTED** [review]
A little late but LGTM. Thanks!

**Pull request merged**